tests/test_tag.py

```diff
@@ -52,77 +49,71 @@ def test(self):
+        self._caplog.clear()
 
-            ret = main(["tag", "list"])
-            self.assertEqual(ret, 0)
+        ret = main(["tag", "list"])
+        self.assertEqual(ret, 0)
 
-            self.assertEqual(
-                logger.handlers[0].stream.getvalue(),
-                "bar.dvc:\n"
-                "  bar:\n"
-                "    v1:\n"
-                "      md5: 8978c98bb5a48c2fb5f2c4c905768afa\n"
-                "foo.dvc:\n"
-                "  foo:\n"
-                "    v1:\n"
-                "      md5: acbd18db4cc2f85cedef654fccc4a4d8\n"
-                "\n",
-            )
+        expected = (
+            "bar.dvc:\n"
+            "  bar:\n"
+            "    v1:\n"
+            "      md5: 8978c98bb5a48c2fb5f2c4c905768afa\n"
+            "foo.dvc:\n"
+            "  foo:\n"
+            "    v1:\n"
+            "      md5: acbd18db4cc2f85cedef654fccc4a4d8\n"
+        )
+
+        # TODO: Format `caplog` output to capture only "%(message)s"
+        stdout = "\n".join(record.message for record in self._caplog.records)
+
+        assert expected == stdout
 
         ret = main(["tag", "remove", "v1"])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_standard_output()
+        self._caplog.clear()
 
-            ret = main(["tag", "list"])
-            self.assertEqual(ret, 0)
+        ret = main(["tag", "list"])
+        self.assertEqual(ret, 0)
 
-            self.assertEqual("", logger.handlers[0].stream.getvalue())
+        self.assertEqual("", self._caplog.text)
 
 
 class TestTagAddNoChecksumInfo(TestDvc):
     def test(self):
         ret = main(["run", "-o", self.FOO, "--no-exec"])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_error_output()
+        self._caplog.clear()
 
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
             ret = main(["tag", "add", "v1", "foo.dvc"])
             self.assertEqual(ret, 0)
 
-            self.assertEqual(
-                "Warning: missing checksum info for 'foo'\n",
-                logger.handlers[1].stream.getvalue(),
-            )
+        assert "missing checksum info for 'foo'" in self._caplog.text
 
 
 class TestTagRemoveNoTag(TestDvc):
     def test(self):
         ret = main(["add", self.FOO])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_error_output()
+        self._caplog.clear()
 
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
             ret = main(["tag", "remove", "v1", "foo.dvc"])
             self.assertEqual(ret, 0)
 
-            self.assertEqual(
-                "Warning: tag 'v1' not found for 'foo'\n",
-                logger.handlers[1].stream.getvalue(),
-            )
+        assert "tag 'v1' not found for 'foo'" in self._caplog.text
```

Comment:
assertEqual is not the same as `assert in ` . Please don't silently change tests like this.

Reply:
`caplog` uses its own formatter, so we can't rely on the idea that output should be _equal_, and more when it's passing through a logger. The `Warning: ` and new line is added by the logger itself, the message is only `tag 'v1' not found for 'foo'`, this kind of breaks the boundaries of testing just the `tag` module, also, to make sure it is being logged with a `warning` level, I used the `caplog.at_level` method to ignore levels below `warning`.

To test the integrity of the message, we could use `self._caplog.records` to retrieve the one that was logged and check for its `.message`, this would be more sound but it kindof makes the test a little bit more convoluted than just checking that the message we wanted to show to the user is included on the logs.

---

It looks sketchy at first glance but I changed all the tests manually with integrity on mind :)